Automattic/woocommerce-payments
# Add transactions search bar

Fixes #597 

#### Changes proposed in this Pull Request

* Add transactions search by customer name and email, and order number

#### Testing instructions



1. Run the server version from server/pull/279
2. Build wc-admin from https://github.com/woocommerce/woocommerce-admin/pull/4518
3. Go to the transactions page
4. Click on the search bar and type a customer name or email
5. You should see a list with suggestions and a wildcard search option
![image](https://user-images.githubusercontent.com/7714042/84067718-36f25700-a99e-11ea-9d59-8f364c4d43a8.png)
6. Clicking on a user should return all transactions for that combination of name and email
![image](https://user-images.githubusercontent.com/7714042/84067761-4bceea80-a99e-11ea-9e38-86e60e62d857.png)
7. Clicking on the wildcard search should return all transactions that include the search term
![image](https://user-images.githubusercontent.com/7714042/84067852-7ae55c00-a99e-11ea-8b55-c91531029d1f.png)
8. Add another search term for an order number (exact match)
9. You should see an "Order #" option
![image](https://user-images.githubusercontent.com/7714042/84067925-994b5780-a99e-11ea-8dbb-60c92ac99a15.png)
10. Clicking on that option should add transactions that match that order to the list
![image](https://user-images.githubusercontent.com/7714042/84067957-a6684680-a99e-11ea-9b14-f26cb326eeaa.png)


-------------------
